### ISSUE TITLE
NoMethodError: undefined method `assign_attributes' for nil:NilClass

### ISSUE THREAD

**Issue by kencheeto** [department-of-veterans-affairs/caseflow#8738]
https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/3064/

**Comment by jiahuang** [comment]
Looked up the intake:

```
<AppealIntake id: 122963, detail_id: 1065, detail_type: "Appeal", 
user_id: 1900, veteran_file_number: "...", 
started_at: "2019-01-15 21:20:39", completed_at: "2019-01-15 21:20:53", 
completion_status: "canceled", error_code: nil, type: "AppealIntake", cancel_reason: "system_error", 
cancel_other: nil, completion_started_at: nil>
```

It has a detail id, but no Appeal with that id exists.

**Comment by leikkisa** [comment]
It looks like it was canceled, so we would have deleted the appeal.  This Veteran has a locked rating, so this is an instance where the intake got canceled so that they wouldn't get closed out of the app for a day.

**Comment by jiahuang** [comment]
@leikkisa you're right, I can see a sentry error for the locked rating at around the same time as this error by the same user for the same intake id: https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/3063/events/220754/

> Rating::NilRatingProfileListError: {:message=>{:error_level=>"0", :reject_reason=>"Locked Rating Detected "}}

**Comment by jiahuang** [comment]
Was able to reproduce this locally.

1. user tries to start an intake which throws a locked rating error
2. we mark intake as failed, delete the associated appeal, and show the "something went wrong page" flash
3. However we don't change the page which allows the user to just try resubmitting again. There is a detail id associated with the current intake, but no actual detail since we destroyed that. 

![review_request](https://user-images.githubusercontent.com/577629/51344243-0fb78480-1a4d-11e9-98b7-ed1f92602731.gif)

**Comment by jiahuang** [comment]
Fix for this would be a dupe of #7385
